lambdas/ttl-poll-lambda/src/__tests__/apis/scheduled-event-handler.test.ts

```diff
@@ -0,0 +1,181 @@
+import { ScheduledEvent } from 'aws-lambda';
+import { Logger } from 'utils';
+import { mock } from 'jest-mock-extended';
+import { EventDetail, createHandler } from 'apis/scheduled-event-handler';
+import { TtlExpiryService } from 'infra/ttl-expiry-service';
+import { ProcessingStatistics } from 'infra/types';
+
+const malformedEvents = [
+  {
+    detail: {
+      dateOfExpiry: 2_323_231,
+    },
+  } as unknown as ScheduledEvent<EventDetail>,
+  {
+    detail: {
+      dateOfExpiry: 'hello',
+    },
+  },
+  {
+    detail: {
+      dateOfExpiry: '02-02-2023',
+    },
+  },
+  {
+    detail: {
+      dateOfExpiry: '2023/02/02',
+    },
+  },
+] as ScheduledEvent<EventDetail>[];
+
+jest.useFakeTimers();
+
+const logger = mock<Logger>();
+jest.spyOn(logger, 'error');
+jest.spyOn(logger, 'info');
+
+jest.mock('../../infra/config', () => ({
+  loadConfig: jest.fn().mockReturnValue({ concurrency: 60 }),
+}));
+
+jest.mock('../../infra/ttl-expiry-service', () => ({
+  ...jest.requireActual('../../infra/ttl-expiry-service'),
+  processExpiredTtlRecords: jest.fn(),
+}));
+
+describe('createHandler', () => {
+  const mockProcessExpiredTtlRecords = jest.fn();
+
+  const mockTtlExpiryService = {
+    processExpiredTtlRecords: mockProcessExpiredTtlRecords,
+  } as unknown as TtlExpiryService;
+
+  const handler = createHandler({
+    logger,
+    ttlExpiryService: mockTtlExpiryService,
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.setSystemTime(new Date('2020-01-01T12:00:00'));
+  });
+
+  it.each(malformedEvents)(
+    'throws error when receiving event %j where dateOfExpiry attribute is not in recognised date format - YYYY-MM-DD',
+    async (event) => {
+      await expect(handler(event)).rejects.toThrow(
+        `dateOfExpiry is not valid ISO date format (YYYY-MM-DD): ${event.detail.dateOfExpiry}`,
+      );
+    },
+  );
+
+  it('throws error when receiving event where timeOfExpiry attribute is defined and cannot be parsed as a date object', async () => {
+    await expect(
+      handler(<ScheduledEvent<EventDetail>>{
+        detail: { dateOfExpiry: '2022-03-07', timeOfExpiry: 'not_a_date' },
+      }),
+    ).rejects.toThrow(
+      `timeOfExpiry is not valid date format (YYYY-MM-DD'T'HH:MM:SSZ): not_a_date`,
+    );
+  });
+
+  it('should validate dateOfExpiry if provided in event payload', async () => {
```

Comment:
I don't think this name correctly describes what this test is verifying.
```suggestion
  it('should use dateOfExpiry if provided in event payload', async () => {
```